src/modules/file-share/create-volume/create-volume.js

```diff
@@ -0,0 +1,496 @@
+                    vm.sendglusterBrickMappingPostReq();
+                }
+
+            } else if (step === "dec") {
+                vm.step -= 1;
+                if (vm.step === 4) {
+                    vm.settings.step = 1;
+                }
+            }
+        }
+
+        /* Step-3: Expand and collapse the Brick Mapping table */
+        function expandList(item) {
+            if (item.isExpanded) {
+                item.isExpanded = false;
+            } else {
+                item.isExpanded = true;
+            }
+        }
+
+        /* Step-4: updating the protocol value in
+        advance settings*/
+        function updateProtocol(protocol) {
+            vm.settings.protocol = protocol;
+        }
+
+        /* Step-4: updating the Transport Type value in
+        advance settings*/
+        function updateTransportType(type) {
+            var typeIndex;
+
+            typeIndex = vm.settings.transportType.indexOf(type);
+
+            if (typeIndex === -1) {
+                vm.settings.transportType.push(type);
+            } else {
+                vm.settings.transportType.splice(typeIndex, 1)
+            }
+        }
+
+        /* Invoked when task is submitted to show the
+        Task submitted page */
+        function _createVolume() {
+            vm.taskSubmitted = true;
+        }
+
+        /* Returns bool value if value for protocol is equal */
+        function isSelectedProtocol(protocol) {
+            return protocol === vm.settings.protocol;
+        }
+
+        /* Return bool value  */
+        function isSelectedType(type) {
+            return vm.settings.transportType.indexOf(type) > -1;
+        }
+
+        /* Takes you to task progress page after
+        submitting the create Volume Job */
+        function viewTaskProgress() {
+            $state.go("task-detail", { taskId: vm.jobId });
+        }
+
+        /* Cancel the Volume creation wizard and
+        takes you to Volume List page */
+        function cancelCreate() {
+            $state.go("file-share");
+        }
+
+        /* TaskProgress: This function will be invoked
+        when you want to create a another Volume */
+        function goToVolumeCreateView() {
+            $state.go("create-volume", {}, { reload: true });
+        }
+
+        /* Step-4: Opens the create brick wizard
+        with the selected cluster */
+        function onCreateBrickModal(clusterData) {
+            var wizardDoneListener,
+                modalInstance,
+                closeWizard;
+
+            modalInstance = $uibModal.open({
+                animation: true,
+                backdrop: "static",
+                templateUrl: "/modules/bricks/create-brick/create-brick.html",
+                controller: "CreateBrickModalController",
+                controllerAs: "vm",
+                size: "lg",
+                resolve: {
+                    clusterData: function() {
+                        return clusterData;
```

Comment:
@cloudbehl We can return a single object which contains both clusterData and selectedCluster.